### PR TITLE
Improve docs and tests for a part of the `setup_run` function

### PR DESCRIPTION
**Pull request by joaomcteixeira**
You are about to submit a new Pull Request. Before continuing make sure you read the [contributing guidelines](CONTRIBUTING.md) and you comply with the following criteria:

- [ ] You have stick to Python. Talk with us before adding other programming languages to HADDOCK3
- [ ] Your PR is about CNS
- [ ] Your code is well documented: proper docstrings and explanatory comments for those tricky parts
- [ ] You structured the code into small functions as much as possible. You can use classes if there's a (state) purpose
- [ ] code follows our coding style
- [ ] You wrote tests for the new code
- [x] `tox` tests pass. *Run `tox` command inside the repository folder*
- [x] `-test.cfg` examples execute without errors. *Inside `examples/` run `python run_tests.py -b`*
- [ ] PR does not add any *install dependencies* unless permission granted by the HADDOCK team
- [ ] PR does not break licensing
- [x] Your PR is about writing documentation for already existing code :fire:
- [x] Your PR is about writing tests for already existing code :godmode:

---

While developing for #382, I found this part of the code poorly documented. Here, I add docs, tests, and correct a minor implementation.

### CODE REVIEW

**Review by amjjbonvin — APPROVED** [review]

File: src/haddock/gear/expandable_parameters.py [hunk 1]
```diff
@@ -396,8 +396,31 @@ def read_mol_parameters(
...
+
+    remove_trail_idx('parameter')
+    >>> 'parameter'
+
+    Parameters
+    ----------
+    param : str
+        A parameter name.
+    """
+    under_parts = param.split("_")
```
Comment (douweschulte) [inline src/haddock/gear/expandable_parameters.py:420]: There is a specialised function for splitting on the last occurrence of a separator.
https://docs.python.org/3/library/stdtypes.html#str.rpartition

File: src/haddock/gear/expandable_parameters.py [hunk 2]
```diff
@@ -396,8 +396,31 @@ def read_mol_parameters(
...
+    if under_parts[-1].isdigit():
+        return "_".join(param.split("_")[:-1])
```
Comment (douweschulte) [inline src/haddock/gear/expandable_parameters.py:422]: Maybe reuse the split here, stored in `under_parts` insetad of splitting again.

File: tests/test_gear_expandable_parameters.py [hunk 3]
```diff
@@ -644,3 +645,21 @@ def test_read_mol_params(inp, default, expected):
...
+    "param, expected",
+    [
+        ("some_parameter_1", "some_parameter"),
+        ("some_parameter_0", "some_parameter"),
+        ("some_parameter_99", "some_parameter"),
+        ("parameter_99", "parameter"),
+        ("parameter99", "parameter99"),
+        ("some_parameter", "some_parameter"),
+        ("parameter", "parameter"),
+        ]
```
Comment (douweschulte) [inline tests/test_gear_expandable_parameters.py:660]: Maybe a nice extra test case would be with a double number `param_0_1`.

File: src/haddock/gear/expandable_parameters.py (hunk 2 above)
Comment (joaomcteixeira) [reply src/haddock/gear/expandable_parameters.py:422]: you are right! what a lapse.

**Comment by douweschulte** [comment]
No problem, I like to help. Please keep asking review I like to read other peoples code and see how problems could be solved.

**Review by douweschulte — APPROVED** [review]

**Review by mgiulini — APPROVED** [review]

**Comment by mgiulini** [comment]
@joaomcteixeira the code looks fine, but a test failed after the last modification, probably because of a network error. can you check it?

**Comment by joaomcteixeira** [comment]
Thanks Marco :+1:

**Pull request merged**